packages/react/src/time-input/TimeInput.test.tsx

```diff
@@ -0,0 +1,132 @@
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+import 'html-validate/jest';
+import { useState } from 'react';
+import { axe, toHaveNoViolations } from 'jest-axe';
+import { cleanup, render, screen } from '../../../../test-utils';
+import { TimeInput, type TimeInputProps, type TimeValue } from './TimeInput';
+
+expect.extend(toHaveNoViolations);
+
+afterEach(cleanup);
+
+type ControlledTimeInputProps = Omit<TimeInputProps, 'label'>;
+
+function renderTimeInput(props?: ControlledTimeInputProps) {
+	return render(<ControlledTimeInput {...props} />);
+}
+
+function ControlledTimeInput({
+	onChange: onChangeProp,
+	value: valueProp,
+	...props
+}: ControlledTimeInputProps) {
+	const [value, setValue] = useState<TimeValue | undefined>(valueProp);
+
+	function onChange(value: TimeValue) {
+		setValue(value);
+		onChangeProp?.(value);
+	}
+
+	return (
+		<TimeInput label="Time" onChange={onChange} value={value} {...props} />
+	);
+}
+
+async function getInput() {
+	const el = await screen.getByRole('textbox');
+	expect(el).toBeInstanceOf(HTMLInputElement);
+	expect(el).toBeInTheDocument();
+	return el as HTMLInputElement;
+}
+
+async function getErrorMessage() {
+	const el = await screen.findByText(errorMessage);
+	expect(el).toBeInstanceOf(HTMLSpanElement);
+	expect(el).toBeInTheDocument();
+	return el as HTMLSpanElement;
+}
+
+const errorMessage = 'Enter a valid time';
+
+describe('TimeInput', () => {
+	it('renders correctly', () => {
+		const { container } = renderTimeInput();
+		expect(container).toMatchSnapshot();
+	});
+
+	it('renders valid HTML with no a11y violations', async () => {
+		const { container } = renderTimeInput();
+		expect(container).toHTMLValidate({
+			extends: ['html-validate:recommended'],
+			// react 18s `useId` break this rule
+			rules: { 'valid-id': 'off' },
+		});
+		expect(await axe(container)).toHaveNoViolations();
+	});
+
+	it('updates correctly based on the `value` prop', async () => {
+		const { container } = renderTimeInput({
+			value: {
+				raw: '2130',
+			},
+		});
+		expect(container).toMatchSnapshot();
+
+		expect(await getInput()).toHaveValue('9:30 pm');
+	});
+
+	it('can render an invalid state', async () => {
+		renderTimeInput({
+			invalid: true,
+			message: errorMessage,
+			value: { raw: 'a' },
+		});
+		expect(await getInput()).toHaveAttribute('aria-invalid', 'true');
```

Comment:
This could just be:

```suggestion
		expect(getInput()).toBeInvalid();
```

More here: https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobeinvalid

Reply:
Thanks